modules/compress/src/main/java/org/apache/ignite/internal/processors/compress/CompressionProcessorImpl.java

```diff
@@ -170,6 +177,28 @@ private ByteBuffer doCompactPage(ByteBuffer page, int pageSize) throws IgniteChe
         return compactPage;
     }
 
+    /** Check if filesystem actually supports punching holes. */
+    private void checkPunchHole(Path storagePath, int fsBlockSz) throws IgniteException {
+        Path testFile = storagePath.resolve("punch_hole_" + UUID.randomUUID() + ".test");
+
+        ByteBuffer buffer = GridUnsafe.allocateBuffer(fsBlockSz * 2);
```

Comment:
Seems we need to call  GridUnsafe.freeBuffer

Reply:
Good catch, thank you